src/groups/mqb/mqbc/mqbc_storageutil.cpp

```diff
@@ -2766,7 +2815,87 @@ int StorageUtil::processCommand(mqbcmd::StorageResult*     result,
                          command.domain().name(),
                          *fileStores);
             return 0;  // RETURN
+        } else if (command.domain().command().isPurgeValue()) {
+            // todo return value
+
+
+            bsl::vector<bsl::vector<mqbcmd::PurgeQueueResult> > purgedQueuesVec;
+            purgedQueuesVec.resize(fileStores->size());
+
+            executeForEachPartitions(
+                bdlf::BindUtil::bind(&purgeDomainDispatched,
+                                    &purgedQueuesVec,
+                                    bdlf::PlaceHolders::_2,  // latch
+                                    bdlf::PlaceHolders::_1,  // partitionId
+                                    *fileStores,
+                                    command.domain().name()),
+                *fileStores);
+
+            mqbcmd::PurgedQueues& purgedQueues = result->makePurgedQueues();
+            for (size_t i = 0; i < purgedQueuesVec.size(); ++i) {
+                const bsl::vector<mqbcmd::PurgeQueueResult>& purgedQs =
+                    purgedQueuesVec[i];
+
+                purgedQueues.queues().insert(purgedQueues.queues().begin(),
+                                            purgedQs.begin(),
+                                            purgedQs.end());
+            }
+
+            return 0;  // RETURN
+        }
+    }
+    else if (command.isQueueValue()) {
+        BSLS_ASSERT_SAFE(command.queue().command().isPurgeAppIdValue());
+
+        const bmqt::Uri uri(command.queue().uri(), allocator);
+
+        BSLS_ASSERT_SAFE(uri.isCanonical());
+
+        mqbi::Storage *queueStorage = NULL;
+        {
+            bslmt::LockGuard<bslmt::Mutex> guard(storagesLock);
```

Comment:
Sent a MSG about `storagesLock`.
Here is an example of its minimal usage.
By the way, it's no harm to access `ReplicatedStorage` without locked `storagesLock`. This lock only sanitaze access to the `vector<map<...>>` container here, and we can safely access `ReplicatedStorage` from the corresponding `FileStore` thread.